internal/corazawaf/body_buffer.go

```diff
@@ -71,9 +108,12 @@ func (br *BodyBuffer) Reader() (io.Reader, error) {
 	return br.writer, nil
 }
 
-// Size returns the current size of the body buffer
-func (br *BodyBuffer) Size() int64 {
-	return br.length
+func (br *BodyBuffer) IsEmpty() bool {
```

Comment:
This is the main API change. Until now we exposed the size so upstream could do checks on the size to determine the action (keep writing, reject when the action beyond the limit is to reject, etc). This is specially problematic because size is constrained to type max value but we could write files larger than that (so programatic correctness) but also because when rejection, we would keep writing the body into temporary files and then decide to drop it on reject.